Documentation/proposals/oci.md

```diff
@@ -0,0 +1,127 @@
+- **fetching**: This subsystem is responsible for downloading images of various types.
+Non-ACI image types (Docker and OCI) are converted to ACI images by delegating to [docker2aci](https://github.com/appc/docker2aci). The logic resides in the `github.com/coreos/rkt/rkt/image` package.
+
+- **image store**: The image store is responsible for persisting and managing downloaded images.
+It consists of two parts, a logic handling a directory tree storing the actual image file blobs (usually residing under `/var/lib/rkt/cas/blob`) and a separate embedded SQL database storing image metadata usually residing in `/var/lib/rkt/cas/db/ql.db`. The implementation resides in the `github.com/coreos/rkt/store/imagestore` package.
+
+- **tree store**: Since dependencies between ACI images form a directed acyclic graph according the [Appc spec](https://github.com/appc/spec/blob/master/spec/ace.md#filesystem-setup) they are pre-rendered in a directory called the tree store cache.
+If the [overlay filesystem](https://www.kernel.org/doc/Documentation/filesystems/overlayfs.txt) is enabled the pre-rendered image is used as the `lowerdir` for the final pod. The implementation resides in the `github.com/coreos/rkt/store/treestore` package.
+
+The actual internal lifecycle of an image is documented in the [architecture documentation](https://github.com/coreos/rkt/blob/master/Documentation/devel/architecture.md#image-lifecycle).
+
+The following table gives an overview of the relevant differences between OCI and Appc regarding image handling aspects:
+
+ Aspect | OCI | Appc image format
+--------|-----|------------------
+Dependencies | Layers array in the [image manifest](https://github.com/opencontainers/image-spec/blob/master/manifest.md#image-manifest) | [Dependency graph](https://github.com/appc/spec/blob/master/spec/ace.md#filesystem-setup)
+Runtime | Separate [image manifest](https://github.com/opencontainers/image-spec/blob/master/manifest.md#image-manifest) and [runtime specification](https://github.com/opencontainers/runtime-spec/blob/master/spec.md) | Single [manifest](https://github.com/appc/spec/blob/master/spec/pods.md#pod-manifest-schema)
+Hash algorithms | Multiple [algorithms](https://github.com/opencontainers/image-spec/blob/master/descriptor.md#algorithms), dominantly SHA-256 | [SHA-512](https://github.com/appc/spec/blob/master/spec/types.md#image-id-type)
+
+Current ongoing work to support OCI in rkt is tracked in the following Github project: https://github.com/coreos/rkt/projects/4
+
+### Goals, non-Goals
+
+With the deprecation of the Appc Spec (https://github.com/appc/spec#-disclaimer-) the current internal rkt architecture is not favorable any more.
+Currently rkt does support ACI, Docker, and OCI images, but the conversion step from OCI to ACI using `docker2aci` seems unnecessary.
+It introduces CPU and I/O bound overhead and is often a source of bugs. For these reasons native support of OCI images inside rkt is envisioned.
```

Comment:
I would say "and is bound by semantical differences between the formats" or something similar.